modules/javafx.controls/src/test/java/test/javafx/scene/control/TableCellTest.java

```diff
@@ -402,6 +415,137 @@ public void testCellInUneditableColumnIsNotEditable() {
+        editingColumn.setOnEditCancel(events::add);
+        table.edit(-1, null);
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTablePosition());
+    }
+
+    /**
+     * Passes before fix of JDK-8187229, must pass after.
+     */
+    @Test
+    public void testEditCancelEventAfterCellReuse() {
+        setupForEditing();
+        int editingIndex = 1;
+        cell.updateIndex(editingIndex);
+        table.edit(editingIndex, editingColumn);
+        TablePosition<?, ?> editingPosition = table.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        cell.updateIndex(0);
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTablePosition());
+    }
+
+    /**
+     * Fails before fix of JDK-8187229.
+     */
+    @Test
+    public void testEditCancelEventAfterModifyItems() {
+        setupForEditing();
+        stageLoader = new StageLoader(table);
+        int editingIndex = 1;
+        table.edit(editingIndex, editingColumn);
+        TablePosition<?, ?> editingPosition = table.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        table.getItems().add(0, "added");
+        Toolkit.getToolkit().firePulse();
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTablePosition());
+    }
+
+    /**
+     * Test that removing the editing item implicitly cancels an ongoing
+     * edit and fires a correct cancel event.
+     */
+    @Test
+    public void testEditCancelEventAfterRemoveEditingItem() {
+        setupForEditing();
+        stageLoader = new StageLoader(table);
+        int editingIndex = 1;
+        table.edit(editingIndex, editingColumn);
+        TablePosition<?, ?> editingPosition = table.getEditingCell();
+        List<CellEditEvent<?, ?>> events = new ArrayList<>();
+        editingColumn.setOnEditCancel(events::add);
+        table.getItems().remove(editingIndex);
+        Toolkit.getToolkit().firePulse();
+        assertNull("sanity: editing terminated on items modification", table.getEditingCell());
+        assertEquals("column must have received editCancel", 1, events.size());
+        assertEquals("editing location of cancel event", editingPosition, events.get(0).getTablePosition());
+    }
+
+    /**
+     * Sanity test: fix must not introduce a memory leak.
+     * Note: not an issue for Tree/TableCell because we store the Tree/TablePosition which
+     * takes care of wrapping everything into weakRefs.
+     */
+    @Test
+    public void testEditCancelMemoryLeakAfterRemoveEditingItem() {
```

Comment:
I took a while to understand this test. 
Maybe one more sentence with information would be helpful here (and same test in TreeTableCell).

(My first thought was we are testing basically WeakReference, but then I figured out we want to test whether there are some references left somewhere which prevents the GC or not. So a small sentence like this would be nice in my opinion :))

Reply:
actually, I went the opposite direction: aligned all method comments to those in TreeCellTest (which boiled down to removing nearly all :)